modules/tinymce/src/plugins/lists/main/ts/core/Delete.ts

```diff
@@ -209,6 +209,15 @@ const backspaceDeleteIntoListCaret = (editor: Editor, isForward: boolean): boole
     const rng = ListRangeUtils.normalizeRange(editor.selection.getRng());
     const otherLi = dom.getParent(findNextCaretContainer(editor, rng, isForward, root), 'LI', root);
 
+    const findValidElement = (element: SugarElement<Node>) => Arr.contains([ 'td', 'th', 'caption' ], SugarNode.name(element));
+    const findRoot = (node: SugarElement<Node>) => node.dom === root;
+    const otherLiCell = PredicateFind.closest(SugarElement.fromDom(otherLi), findValidElement, findRoot);
+    const caretCell = PredicateFind.closest(SugarElement.fromDom(rng.startContainer), findValidElement, findRoot);
+
+    if (otherLiCell !== caretCell) {
```

Comment:
This can't be right since `PredicateFind.closest` returns a Optional. Use something like `Optionals.equals`. Also since this is not breaking any test that suggest that more tests are needed.

- [ ] Switch to proper equals